src/dataprotection/azext_dataprotection/manual/custom.py

```diff
@@ -139,6 +139,61 @@ def dataprotection_backup_instance_list_from_resourcegraph(client, datasource_ty
     return response.data
 
 
+def dataprotection_backup_instance_update_msi_permissions(cmd, client, resource_group_name, datasource_type, vault_name, operation, permissions_scope, backup_instance=None):
+    backup_vault = client.get(resource_group_name=resource_group_name,
+                vault_name=vault_name)
+    principal_id = backup_vault.identity.principal_id
+
+    role_assignments_arr = []
+
+    if operation == 'Backup' and backup_instance is None:
+        raise CLIError("Backup instance needs to be provided when operation is Backup")
+
+    from azure.cli.command_modules.role.custom import list_role_assignments, create_role_assignment
+
+    if datasource_type == 'AzureDisk':
+        # Create scope variables for role assignment
+        resource_scope_assignment = backup_instance['properties']['data_source_info']['resource_id']
+        snapshot_rg_scope_assignment = backup_instance['properties']['policy_info']['policy_parameters']['data_store_parameters_list'][0]['resource_group_id']
+        resource_scope = resource_scope_assignment
+        snapshot_rg_scope = snapshot_rg_scope_assignment
+
+        if permissions_scope == 'Resource Group':
+            resource_scope_assignment = "/".join(resource_scope_assignment.split("/")[:5]) # Snapshot RG is already in RG scope, so change for resource
+        elif permissions_scope == 'Subscription':
+            # Change scope for both resource and Snapshot RG
+            resource_scope_assignment = "/".join(resource_scope_assignment.split("/")[:3])
+            snapshot_rg_scope_assignment = "/".join(snapshot_rg_scope_assignment.split("/")[:3])
+        
+        
+        role_assignments = [obj['roleDefinitionName'] for obj in list_role_assignments(cmd, assignee=principal_id, scope=resource_scope, include_inherited=True)]
```

Comment:
Better naming for local variables can be used. like `role_assignment` instead of a general name like `obj`